delaymodel.py

```diff
@@ -229,64 +231,83 @@ def calculate_delay(self, publish : bool = True):
-                # delay2 = -delay2
-                # delay3 = -delay3
-
-                # Compute the delay rate in s/s
-                rate1 = (delay2 - delay1) / (dt)
-                rate2 = (delay3 - delay2) / (dt)
-                rate = (delay3 - delay1) / (2*dt)
-
-                # Compute the delay rate rate in s/s^2
-                raterate = (rate2 - rate1) / (dt)
-
-                self.delay_data["delay_ns"] = (delay2*1e9).tolist()
-                self.delay_data["delay_rate_nsps"] = (rate*1e9).tolist()
-                self.delay_data["delay_raterate_nsps2"] = (raterate*1e9).tolist()
-                self.delay_data["effective_lo_0_mhz"] = self.lo_eff[0]
-                self.delay_data["effective_lo_1_mhz"] = self.lo_eff[1]
-                self.delay_data["sideband_0"] = self.sideband[0]
-                self.delay_data["sideband_1"] = self.sideband[1]
-                self.delay_data["time_value"] = tts[1]
-
-                if publish:
-                    self.publish_delays()
-                else:
-                    return pd.DataFrame(self.delay_data, index=list(self.itrf.index.values)).to_dict('index')
-
-                #sleep for at least 2 seconds
-                time.sleep(2)
+
+            t = time.time()
+            t_int = np.floor(t)
+            loadtime_from_now = self.source_pointings_dict['loadtime'] - t
+
+            if (loadtime_from_now < self.min_loadtime_offset
+                or 
+                loadtime_from_now > self.max_loadtime_offset):
+                #loadtime for source pointing is too far in the future or is in the past. Use current pointings on hand and set loadtime to 
+                #self.min_loadtime_offset from now.
+                loadtime = t_int + self.min_loadtime_offset
+
+            elif(loadtime_from_now >= self.min_loadtime_offset
+                and
+                loadtime_from_now <= self.max_loadtime_offset):
+                #loadtime is within threshold. Update source pointing and calculate coefficients and set loadtime to 
+                #loadtime.
+                self.source = SkyCoord(self.source_pointings_dict['ra'], self.source_pointings_dict['dec'], unit='deg')
+                loadtime = self.source_pointings_dict['loadtime']
+                sideband = self.source_pointings_dict['sideband']
+                sslo = self.source_pointings_dict['sslo']
+
+            elif loadtime_from_now is None:
+                #loadtime is None. Update source pointing and calculate coefficients and set loadtime to 
+                #self.min_loadtime_offset from now.
+                self.source = SkyCoord(self.source_pointings_dict['ra'], self.source_pointings_dict['dec'], unit='deg')
+                loadtime = t_int + self.min_loadtime_offset
+                sideband = self.source_pointings_dict['sideband']
+                sslo = self.source_pointings_dict['sslo']
+            
+            else:
+                logger.error("Invalid loadtime provided")
+
```

Comment:
I assume everything below here is as it was, just with reduced indenting

Reply:
Yeah you're right, had an additional `if` statement present which I removed.